Lib/asyncio/base_events.py

```diff
@@ -647,6 +647,66 @@ def _getaddrinfo_debug(self, host, port, family, type, proto, flags):
         return await self.run_in_executor(
             None, socket.getnameinfo, sockaddr, flags)
 
+    async def sock_sendfile(self, sock, file, offset=0, count=None,
+                            *, fallback=True):
+        if self._debug and sock.gettimeout() != 0:
+            raise ValueError("the socket must be non-blocking")
+        self._check_sendfile_params(sock, file, offset, count)
+        try:
+            return await self._sock_sendfile_native(sock, file,
+                                                    offset, count)
+        except RuntimeError:
```

Comment:
many things in sock_sendfile_native can cause a RuntimeError that this code will mask. I recommend to just add `hasattr(os, 'sendfile')`

Reply:
Or add a private exception class (not exported and with a leading underscore)

Reply:
Interesting migration.
Private exception class -> NotImplementedError -> RuntimeError -> private exception again.
I'm ok with custom exception, it is the best choice.

Reply:
Sorry, Andrew ;) Sometimes it's hard to see the big picture when we just start working on a new big PR

Happy New Year, by the way!!